src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1650,6 +1670,10 @@ void TorrentHandle::handleTorrentFinishedAlert(libtorrent::torrent_finished_aler
             forceRecheck();
         m_session->handleTorrentFinished(this);
     }
+
+    if (hasPauseOnCompletion()) {
```

Comment:
Handling torrent finished event is too complex job. At first glance, pause in this place prevents finished torrent rechecking.

Reply:
Where should it be paused?